dashboard/app/controllers/regional_partners_controller.rb

```diff
@@ -0,0 +1,79 @@
+class RegionalPartnersController < ApplicationController
+  authorize_resource class: :pd_regional_partner_management
+  before_action :set_regional_partner, only: [:show, :edit, :update]
+
+  # GET /regional_partners
+  def index
+    search_term = params[:search_term]
+    if search_term =~ /^\d+$/
+      @regional_partners = RegionalPartner.where(id: search_term)
+    elsif search_term
+      @regional_partners = RegionalPartner.where("name LIKE :partial_name", {partial_name: "%#{search_term}%"})
+    end
+
+    unless @regional_partners.try(:any?) || search_term.blank?
+      flash[:notice] = "No matching Regional Partners found. Showing all."
+    end
+    # display all regional partners if search does not return any results
+    @regional_partners = RegionalPartner.all unless @regional_partners.try(:any?)
+  end
+
+  # GET /regional_partners/:id
+  def show
+  end
+
+  # GET /regional_partners/new
+  def new
+    @regional_partner = RegionalPartner.new
```

Comment:
note `load_resource` or `load_and_authorize_resource` (suggested above) will handle this for you